git-history.md

```diff
@@ -0,0 +1,69 @@
+# Git History Guidelines
+
+All of our projects follow a *fork + feature branch workflow*. That is, every
+project has a main *upstream* repository (e.g.
+[in-toto/in-toto](https://github.com/in-toto/in-toto)), with a *main* branch,
+called `master` or `develop` (e.g.
+[in-toto/in-toto@develop](https://github.com/in-toto/in-toto/tree/develop)).
+Contributors work on their own *fork* of the *upstream* repository (e.g.
+[lukpueh/in-toto](https://github.com/lukpueh/in-toto)). For every feature or
+fix they are working on, they create a *feature* branch, with a descriptive
+name (e.g.
+[lukpueh/in-toto@threshold-comment-fix](https://github.com/lukpueh/in-toto/tree/threshold-comment-fix)).
+A *feature* branch of a *fork* is incorporated into the *main* branch of the
+*upstream* repository by submitting a *pull request*.
+
+The rest of this document presents some guidelines to keep our projects' git
+histories clean.
+
+__*TL;DR*__
+Avoid merge commits in your *feature* branch. You can use `git rebase` to sync
+your *feature* branch with the *main* branch in *upstream* and also to keep
+your git history clean.
+
+
+__Choose the right base for your *feature* branch__
+In most cases you will base your *feature* branch on the *main* branch of your
+*fork*. Make sure that the *main* branch of your fork is in sync with the
+*main* branch of *upstream* so that you don't work on outdated code.
+Updating the *main* branch of your fork should not require *merge* commits,
+unless you have commited directly to your *main* branch, which you shouldn't!
+
+__Keep the base of your *feature* branch in sync with *upstream*__
+While you are working on your *feature* branch the *main* branch might move
+forward, because some other work was merged into the *main* branch. Check
+regularily, especially before you submit a *pull request*, if the
+base of your *feature* branch has moved forward. If it has, use `git rebase`
```

Comment:
[*GitHub and rebase test*](https://github.com/secure-systems-lab/lab-guidelines/pull/14/files#r235340237) 

Testing a review comment on a line that **will not change** and links to lines that change and lines that don't change after a rebase and force push.

1. https://github.com/secure-systems-lab/lab-guidelines/pull/14/files#diff-b4039e218e46891be5b013b445f12c47R37
1. https://github.com/secure-systems-lab/lab-guidelines/blame/84b06d128cff13ecadc017793d7f41bb8001dcf8/git-history.md#L37
1. https://github.com/secure-systems-lab/lab-guidelines/blob/84b06d128cff13ecadc017793d7f41bb8001dcf8/git-history.md
1. https://github.com/secure-systems-lab/lab-guidelines/pull/14/files#diff-b4039e218e46891be5b013b445f12c47R36
1. https://github.com/secure-systems-lab/lab-guidelines/blame/84b06d128cff13ecadc017793d7f41bb8001dcf8/git-history.md#L36